tests/csapi/account_deactivate_test.go

```diff
@@ -16,6 +20,41 @@ func TestDeactivateAccount(t *testing.T) {
 	password := "superuser"
 	authedClient := deployment.RegisterUser(t, "hs1", "test_deactivate_user", password, false)
 	unauthedClient := deployment.Client(t, "hs1", "")
+
+	// Ensure that the first step, in which the client queries the server's user-interactive auth flows, returns
+	// at least one auth flow involving a password.
+	t.Run("Password flow is available", func(t *testing.T) {
+		reqBody := client.WithJSONBody(t, map[string]interface{}{})
+		res := authedClient.DoFunc(t, "POST", []string{"_matrix", "client", "v3", "account", "deactivate"}, reqBody)
+		must.MatchResponse(t, res, match.HTTPResponse{
+			StatusCode: 401,
+			JSON: []match.JSON{
+				match.JSONArrayEach(
+					"flows",
+					func(flow gjson.Result) error {
+						stages := flow.Get("stages")
+						if !stages.IsArray() {
+							return fmt.Errorf("flows' stages is not an array.")
+						}
+						foundStage := false
+						for _, stage := range stages.Array() {
+							if stage.Type != gjson.String {
+								return fmt.Errorf("stage is not a string")
+							}
+							if stage.Str == "m.login.password" {
+								foundStage = true
+							}
+						}
+						if !foundStage {
+							return fmt.Errorf("No m.login.password stage found.")
+						}
+						return nil
+					},
+				),
+			},
```

Comment:
I think this is going to test that every flow contains an `m.login.password` stage.

I think we instead want to test that at least one flow contains an `m.login.password` stage?

Reply:
Though having said that: must a homeserver offer password-based authentication at all? I'm not sure that's a requirement of the spec!

Maybe it'd be easier to assert that
- we get a response object containing at least one flow
- each flow consists of >= 1 stages

Would that have detected the Synapse screwup? (I think even the status code check would suffice, but...)